extension/js/agent/patches/patchBackboneView.js

```diff
@@ -11,49 +12,55 @@
         });
       }
     });
-  }, 200);
-
-  // @private
-  Agent.patchBackboneView = function(BackboneView) {
-    debug.log('Backbone.View detected');
+  };
 
-    Agent.patchBackboneComponent(BackboneView, function(view) { // on new instance
-      Agent.lazyWorker.push({
-        context: Agent,
-        args: [view],
-        callback: function(view) {
-          // registra il nuovo componente dell'app
-          var data = Agent.serializeView(view)
-          Agent.registerAppComponent('View', view, data);
-          Agent.sendAppComponentReport('view:change', {
-            data: data,
-            cid: view.cid
-          });
-        }
-      });
-
-      // set class name if we can
-      if (view._requirePath) {
-        view._className = _.last(view._requirePath.split('/'));
+  var onNewViewInstance = function(view) { // on new instance
+    debug.log('new view instance:', view.cid);
+    // flag to not trigger view:change when is registering
+    var isRegisteringView = true;
+    var viewChanged = _.debounce(_.partial(patchViewChanges, view), 0);
+    Agent.lazyWorker.push({
+      context: Agent,
+      args: [view],
+      callback: function(view) {
+        // registra il nuovo componente dell'app
+        var data = Agent.serializeView(view);
+        Agent.registerAppComponent('View', view, data);
       }
+    });
 
-      // Patcha i metodi del componente dell'app
-      Agent.patchAppComponentTrigger(view, 'view');
+    // set class name if we can
+    if (view._requirePath) {
+      view._className = _.last(view._requirePath.split('/'));
+    }
 
-      Agent.onDefined(view, 'ui', function() {
-        Agent.onChange(view.ui, _.partial(patchViewChanges, view));
-        patchViewChanges(view);
-      });
+    // Patcha i metodi del componente dell'app
+    Agent.patchAppComponentTrigger(view, 'view');
 
-      Agent.onDefined(view, '_events', function() {
-        Agent.onChange(view._events, _.partial(patchViewChanges, view));
-        patchViewChanges(view);
-      });
+    Agent.onDefined(view, 'ui', function() {
+      Agent.onChange(view.ui, viewChanged);
+      if (!isRegisteringView) viewChanged();
+    });
+
+    Agent.onDefined(view, '_events', function() {
+      Agent.onChange(view._events, viewChanged);
+      if (!isRegisteringView) viewChanged();
+    });
+
+    Agent.onDefined(view, '_listeningTo', function() {
+      Agent.onChange(view._listeningTo, viewChanged);
+      if (!isRegisteringView) viewChanged();
+    });
+
+    isRegisteringView = false;
+  };
+
+  // @private
+  Agent.patchBackboneView = function(BackboneView) {
+    debug.log('Backbone.View detected');
 
-      Agent.onDefined(view, '_listeningTo', function() {
-        Agent.onChange(view._listeningTo, _.partial(patchViewChanges, view));
-        patchViewChanges(view);
-      });
+    Agent.patchBackboneComponent(BackboneView, function (view) {
+      _.defer(onNewViewInstance, view);
```

Comment:
why defer?

Reply:
This event is called before BB.View initialize. Later some monitored properties (_listeningTo, ui, _events) are commonly defined, triggering onDefined events. By calling `onNewViewInstance` after those properties are defined and checking for `isRegisteringView` it avoids sending redundant 'view:change' messages and the view serialization costs